clients/python-pyo3/tests/test_client.py

```diff
@@ -0,0 +1,1032 @@
+# type: ignore
+"""
+Tests for the TensorZero client
+
+We use pytest and pytest-asyncio to run the tests.
+
+These tests cover the major functionality of the client but do not
+attempt to comprehensively cover all of TensorZero's functionality.
+See the tests across the Rust codebase for more comprehensive tests.
+
+To run:
+```
+pytest
+```
+or
+```
+uv run pytest
+```
+"""
+
+from copy import deepcopy
+from enum import Enum
+from time import time
+from uuid import UUID
+
+import pytest
+import pytest_asyncio
+from tensorzero import (
+    AsyncTensorZeroGateway,
+    ChatInferenceResponse,
+    FeedbackResponse,
+    JsonInferenceResponse,
+    TensorZeroError,
+    TensorZeroGateway,
+    Text,
+    ToolCall,
+    ToolResult,
+)
+from uuid_utils import uuid7
+
+
+class ClientType(Enum):
+    HttpGateway = 0
+    EmbeddedGateway = 1
+
+
+@pytest_asyncio.fixture(params=[ClientType.HttpGateway, ClientType.EmbeddedGateway])
+async def async_client(request):
+    if request.param == ClientType.HttpGateway:
+        async with AsyncTensorZeroGateway("http://localhost:3000") as client:
+            yield client
+    else:
+        async with await AsyncTensorZeroGateway.create_embedded_gateway(
+            config_path="/home/aaron/repos/tensorzero/tensorzero_internal/tests/e2e/tensorzero.toml",
```

Comment:
gotta use a relative path here

Reply:
Oops, I thought I changed this before committing :laughing: